website/content/docs/api-gateway.mdx

```diff
@@ -0,0 +1,281 @@
+| `-sds-server-host`     | Integer value that specifies the port number for the secret discovery service (SDS).                                                                                                  | Optional | `9090`                                                                  |
+
+You can also issue the `version` command to print the Consul API Gateway version to the console:
+
+```shell-session
+$ ./consul-api-gateway version
+consul-api-gateway 0.1.0
+```
+--->
+
+## Configuration
+
+Configure the following artifacts to facilitate ingress into your Consul service mesh:
+
+- [Gateway](#gateway): Defines the main infrastructure resource that links API gateway components. It specifies the name of the `GatewayClass` and one or more `listeners`.
+- [GatewayClass](#gatewayclass): Defines a class of gateway resources that you can use as a template for creating gateways.
+- [GatewayClassConfig](#gatewayclassconfig): Describes additional Consul API Gatway-related configuration parameters for the `GatewayClass` resource.
+- [Listeners](#listeners): Defines listener properties, such as protocol, port, and namespace.
+- [Routes](#routes): Specifies the path from the client to the listener.
+
+### Gateway
+
+The gateway configuration is the main infrastructure resource that links API gateway components. It specifies the name of the `GatewayClass` and one or more `listeners`.
+
+Add the `kind: Gateway` option to the configuration file to declare a gateway.
+The following example creates a gateway called `example-gateway`.
+The gateway is based on the `test-gateway-class` and includes a listener called `https` (see [Listeners](#listeners) for details about the `listener` configuration).
+
+<CodeBlockConfig filename="gateway.yaml" lineNumbers highlight="2,4,8,9,13">
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: Gateway
+metadata:
+  name: example-gateway
+  annotations:
+    'external-dns.alpha.kubernetes.io/hostname': DNS_HOSTNAME
+spec:
+  gatewayClassName: test-gateway-class
+  listeners:
+    - protocol: HTTPS
+      hostname: DNS_HOSTNAME
+      port: 443
+      name: https
+      allowedRoutes:
+        namespaces:
+          from: Same
+      tls:
+        certificateRefs:
+          - name: gateway-production-certificate
+```
+
+</CodeBlockConfig>
+
+Refer to the Kubernetes Gateway API documentation for details about configuring gateways:
+<https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.Gateway>
+
+### GatewayClass
+
+The `GatewayClass` resource is used as a template for creating `Gateway` resources.
+The specification includes the name of the controller (`controllerName`) and an API object containing controller-specific configuration resource within the cluster (`parametersRef`).
```

Comment:
We may want to add that the value `'hashicorp.com/consul-api-gateway-controller'` is required for `controllerName`, as that is what tells our controller we own the `Gateway` objects associated with this `GatewayClass`.